lua/nvim-lsp-installer/middleware.lua

```diff
@@ -25,6 +26,9 @@ function M.register_lspconfig_hook()
         local ok, server = servers.get_server(config.name)
         if ok then
             merge_in_place(config, server._default_options)
+            if settings.current.automatic_installation and not server:is_installed() then
+                server:install()
```

Comment:
This is not blocking, isn't it? So installing would take some time for downloading and unpacking, say 10 seconds. But this hook will be executed when lspconfig.setup is called. How do we ensure server is ready and available at this moment?

Reply:
Nope, it's not blocking!

>  How do we ensure server is ready and available at this moment, or lspconfig[...].setup is scheduled to be invoked later after the installation is complete?

We don't, anymore. It's one of the drawbacks, but I'd almost argue that's now a usability issue with lspconfig?

Reply:
Yeah, so my understanding is correct. Actually this was to point out some potential problems in the logical flow rather than just questions --- so even if we do `server:install()`, the installation won't be complete until the actual `lspconfig[..].setup { }` call that immediately follows this `on_setup` hook. Therefore the missing LSP still not work as expected this time (despite `automatic_installation`), with a hope that the missing LSP will work next time, which can be a bit confusing to users. I know there's nothing much we can do here other than this, but at least there could be some notifications or documentation about the timing and behavior.

Reply:
Yeah I want to improve this, will let the new things settle for a while first!